src/features/formData/formDataSlice.ts

```diff
@@ -112,7 +112,7 @@ export const formDataSlice = () => {
           } else {
             state.formData[field] = data;
           }
-          if (!skipAutoSave) {
+          if (skipAutoSave) {
             state.unsavedChanges = true;
```

Comment:
Im not sure the logikk behinde this either.

Reply:
When you make changes to the data model server-side (in the `ProcessDataWrite` function), we pass on the changes with `skipAutoSave = true` to avoid us automatically sending the data to the server again (and creating an infinite save loop, potentially). See the `handleChangedFields` function for the implementation.